### DISCUSSION TITLE
Document Upload/Upsert in Share Chatbot

### DISCUSSION THREAD

**Discussion by iyinusa** [FlowiseAI/Flowise#1975] [Ideas]
I am trying to prototype and share the chatbot to enable seamless experience and feedback. However, the Share Chatobot feature is so lovely that it takes the testing experience away for the architectural canvas, but the document embedding, I feel the experience is missing the piece to enable document upload from the chat interface.

Adding the upload and seamless upsert feature for embedding flows will be great.

What do you think?

**Comment by roscoe1912** [comment]
agreed

**Comment by HenryHengZJ** [comment]
Currently alot of users build their own frontend that allow end users to upload files, and behind the scene it calls the `/vector/upsert` API. If we have more capacity, we would love to work on that as well!

**Reply by iyinusa** [reply]
Awesome! Yes, true, I also later figured it out, after checking the API, which is very cool!
But for the purpose of prototype testing, it will be cool.

Thanks for the response.
